src/main/java/org/terasology/launcher/repositories/JenkinsClient.java

```diff
@@ -45,23 +49,28 @@
      * @return an input stream similar to {@code url.openStream()} with a connection timeout of 10 seconds
      * @throws IOException if opening the URL connection or retrieving the input stream fails
      */
-    static InputStream openStream(URL url) throws IOException {
+    static InputStream openStream(URL url) throws IOException, URISyntaxException, InterruptedException {
         // this is a static member to indicate that it is independent of the client itself, and to cleanly stub it for
         // testing purposes without the need to mock the class-under-test itself.
-        URLConnection connection = url.openConnection();
-        connection.setConnectTimeout(10 * 1000);
-        return connection.getInputStream();
+        var client = HttpClient.newBuilder()
+                .connectTimeout(Duration.ofSeconds(10))
+                .followRedirects(HttpClient.Redirect.NORMAL)
+                .build();
+        var request = HttpRequest.newBuilder(url.toURI()).build();
+        var response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
```

Comment:
Not sure I like those `var`s too much ... :thinking:  When building a client or request it's clear to me what the type most likely is, but I'm less sure what the response actually is. Sure, my IDE does show it to, but unfortunately Github does not :confused: 

I don't think we should be as strict as the [presto community](https://www.lewuathe.com/when-to-use-var-in-java-10.html), as I think that `var` does have it's place for local fields in many places... 
I probably just need to get used to them :sweat_smile: